src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -44,6 +50,80 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
+     * @throws ParseException If the format is invalid or bounds are incorrect.
+     */
+    public static List<Index> parseInterval(String range) throws ParseException {
+        String[] bounds = range.split("-");
+        System.out.println(bounds.toString());
+        // Validate that the range contains exactly two parts
+        if (bounds.length != 2) {
+            throw new ParseException(MESSAGE_INVALID_INTERVAL_FORMAT);
+        }
+
+        // Parse start and end indices
+        Index startIndex = parseIndex(bounds[0]);
+        Index endIndex = parseIndex(bounds[1]);
+
+        // Validate that startIndex is less than or equal to endIndex
+        if (startIndex.getOneBased() > endIndex.getOneBased()) {
+            throw new ParseException(MESSAGE_INVALID_INTERVAL);
+        }
+
+        // Generate and return the list of indices from start to end
+        return IntStream.rangeClosed(startIndex.getOneBased(), endIndex.getOneBased())
+                .mapToObj(Index::fromOneBased)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Parses a string containing indices and ranges (e.g., "1 2 3 5-9") into a list of {@code Index} objects.
+     * Accepts both single indices and ranges separated by spaces.
+     *
+     * @param args The string containing indices and/or ranges to parse.
+     * @return A list of {@code Index} objects representing the parsed indices and ranges.
+     * @throws ParseException If any part of the input is not a valid index or range.
+     */
+    public static List<Index> parseIndices(String args) throws ParseException {
+        requireNonNull(args);
+        String trimmedIndices = args.trim();
+
+        try {
+            return Arrays.stream(trimmedIndices.split(" "))
+                    .flatMap(arg -> parseArgAsStream(arg))
+                    .collect(Collectors.toList());
+        } catch (RuntimeException e) {
+            String causeMessage = e.getCause().getMessage();
+            // Catch any ParseExceptions wrapped in RuntimeExceptions
+            throw new ParseException(causeMessage, e);
+        }
+    }
+
+    /**
+     * Parses a single argument as either an interval (e.g., "5-9") or a single index (e.g., "3"),
+     * returning a stream of {@code Index} objects. This allows for processing both individual indices
+     * and ranges within a stream context.
+     *
+     * @param arg The argument string to parse as an index or interval.
+     * @return A stream of {@code Index} objects parsed from the argument.
+     * @throws RuntimeException wrapping a {@code ParseException} if the argument is invalid.
+     */
+    private static Stream<Index> parseArgAsStream(String arg) {
+        try {
+            if (arg.contains("-")) {
+                return parseInterval(arg).stream();
+            } else {
+                return Stream.of(parseIndex(arg));
+            }
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Just wondering if there is a need to wrap the ParseException in a RuntimeException?

Reply:
This wrapping occurs because Java’s Stream API doesn’t allow checked exceptions like ParseException within its lambda expressions, so the ParseException has to be wrapped in an unchecked exception (RuntimeException) to fit within the stream operations. The code then unwraps and rethrows it to maintain the original context and provide a meaningful ParseException for the caller.

Other requested changes has been pushed